prov/efa/src/rxr/rxr_ep.c

```diff
@@ -2140,14 +2144,14 @@ void rxr_ep_progress_internal(struct rxr_ep *ep)
 			continue;
 
 		if (tx_entry->window > 0)
-			tx_entry->send_flags |= FI_MORE;
+			flags |= FI_MORE;
 		else
 			continue;
 
 		while (tx_entry->window > 0) {
 			if (ep->efa_max_outstanding_tx_ops - ep->efa_outstanding_tx_ops <= 1 ||
 			    tx_entry->window <= ep->max_data_payload_size)
-				tx_entry->send_flags &= ~FI_MORE;
```

Comment:
I think the code can be simplified here. 

The first part of setting `flags` is non necessary. in the loop you can do 
```
flags = FI_MORE
if (xxxx) {
   flags = 0;
}
```
the unset of `FI_MORE` from flags can also be removed. It was there because `tx_entry->send_flags` is persistent.

Reply:
That's a better idea. Thanks, will revise it accordingly.

Reply:
Fixed in latest revision.